src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -43,12 +45,46 @@ public override void Initialize(IHostContext hostContext)
             Trace.Info($"Container options: {container.ContainerCreateOptions}");
             Trace.Info($"Skip container image pull: {container.SkipContainerImagePull}");
 
+            // Check whether we are inside a container.
+#if OS_WINDOWS
+            // service CExecSvc is Container Execution Agent.
+            ServiceController[] scServices = ServiceController.GetServices();
+            if (scServices.Any(x => String.Equals(x.ServiceName, "cexecsvc", StringComparison.OrdinalIgnoreCase) && x.Status == ServiceControllerStatus.Running))
+            {
+                throw new NotSupportedException("The agent process is already inside container.");
```

Comment:
Container feature is not supported when agent is already running inside container. fwlink/aka.

Reply:
https://go.microsoft.com/fwlink/?linkid=875268

Reply:
Can I ask what the reasoning behind it? My Agent runs inside a container into which I mounted docker pipeline and docker executable containing folder, i.e.:
```
-v "C:/Program Files/Docker:C:/Program Files/Docker" `
-v //./pipe/docker_engine://./pipe/docker_engine `
```
So whenever it does anything with Docker - it does it on Docker Host, not inside a container.